lib/tests/streamlit/streamlit_test.py

```diff
@@ -454,6 +453,48 @@ def test_st_pyplot(self):
         checksum = "iVBORw0KGgoAAAANSUhEUgAAAZAAAAGQCAYAAACAvzb"
         self.assertTrue(el.imgs.imgs[0].data.base64.startswith(checksum))
 
+    def test_st_pyplot_clear_figure(self):
+        """st.pyplot should clear the passed-in figure."""
+        if sys.version_info < (3, 0):
+            # We don't test matplotlib under Python 2, because we're not
+            # able to reliably force the backend to "agg".
+            return
+
+        import matplotlib
+        import matplotlib.pyplot as plt
+
+        if matplotlib.get_backend().lower() != "agg":
+            plt.switch_backend("agg")
+
+        # Assert that plt.clf() is called by st.pyplot() only if
+        # clear_fig is True
+        for clear_fig in [True, False]:
```

Comment:
Could this be done with `parametrized`? I see you have a preamble, though, so I guess that may be hard to express.

Reply:
It could probably be done with parametrized, but I think only with extra annoying gymnastics? I'm not sure what it would buy us.